app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -62,10 +62,10 @@ def reload_answer
   def publish_all
     graded_submissions = @assessment.submissions.with_graded_state
     if !graded_submissions.empty?
-      graded_submissions.update_all(
-        workflow_state: 'published', publisher_id: current_user.id, published_at: Time.zone.now,
-        updated_at: Time.zone.now, updater_id: current_user.id
-      )
+      graded_submissions.each do |submission|
```

Comment:
I think all exp can be updated using 1-2 query, it breaks abstraction but we get the performance.

The other way is to put the updating code in a job if it takes too long.